apps/yggdrasil_web/test/controllers/character_controller_test.exs

```diff
@@ -0,0 +1,96 @@
+defmodule YggdrasilWeb.CharacterControllerTest do
+  use YggdrasilWeb.ConnCase
+
+  alias YggdrasilWeb.CharacterView
+  alias Yggdrasil.{Repo, User, Game, Character}
+
+  @user %{username: "tester",
+          password: "password",
+          password_confirmation: "password"}
+
+  @json_api_content_type "application/vnd.api+json"
+  @json_api_content_type_utf8 @json_api_content_type <> "; charset=utf-8"
+
+  setup %{conn: conn} do
+    # need a user to gen a token for api
+    {:ok, user} = %User{}
+    |> User.create_changeset(@user)
+    |> Repo.insert
+
+    #api token
+    {:ok, token, _claims} = Guardian.encode_and_sign user, :token
+
+
+    # set of games to have something to list
+    games = Enum.map 1..2, fn n ->
+      game = Game.changeset(%Game{}, %{"name" => "game#{n}",
+                                       "description" => "game desc #{n}"})
+      {:ok, game} = Repo.insert game
+
+      game
+    end
+
+    chars = Enum.flat_map games, fn g ->
+      Enum.map 1..2, fn n ->
+        char = Character.changeset(%Character{}, %{:name => "char #{g.name} #{n}",
+                                                   :game_id => g.id,
+                                                   :user_id => user.id})
+
+        {:ok, char} = Repo.insert char
+
+        char
+      end
+    end
+
+    conn = conn
+      |> put_req_header("content-type", @json_api_content_type)
+      |> put_req_header("accept", @json_api_content_type)
+
+    {:ok, %{conn: conn, chars: chars, games: games, user: user, token: token}}
+  end
+
+  test "get /api/characters/char_id returns the correct character", ctx do
+    char = Enum.at ctx.chars, 0
+
+    char_json = CharacterView.render("show.json", conn: %{}, data: char)
```

Comment:
After studying how JaSerializer tests their stuff noticed that they just use the view directly like this. 

That said I think JaSerializer has format method that it uses internally that accepts a connection and formater. 

Seeing this example showing that connection can be empty I think we could use it directly and pass the view in as the formatter. Not sure what is best here but this works better than hand coding a solution like I have in other tests. 

I think after this branch I need to make a sweep at our tests and make them more consistent.